From 11a956dfb280dcbae2b10cc66ed6de06be2dfb98 Mon Sep 17 00:00:00 2001
From: Marek Waszkiewicz <mwaszkiewicz@anaconda.com>
Date: Wed, 6 Dec 2023 16:40:13 +0100
Subject: [PATCH] load ansicon library from conda default location

---
 ansicon/__init__.py | 4 +---
 setup.py            | 1 -
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/ansicon/__init__.py b/ansicon/__init__.py
index a4cb8b0..5d5a7a1 100644
--- a/ansicon/__init__.py
+++ b/ansicon/__init__.py
@@ -27,9 +27,7 @@ def _get_dll_path():
     Determines the path of the ANSICON DLL
     """
 
-    directory = os.path.dirname(os.path.abspath(__file__))
-    dll = 'ANSI64.dll' if sys.maxsize > 2**31 else 'ANSI32.dll'
-    return os.path.join(directory, dll)
+    return os.path.join(os.environ['CONDA_PREFIX'], 'Library', 'bin', 'ANSI64.dll')
 
 
 class _Loader(object):
diff --git a/setup.py b/setup.py
index 5e6afd2..f03e6bc 100644
--- a/setup.py
+++ b/setup.py
@@ -40,7 +40,6 @@ setup(
     install_requires=[],
     tests_require=[],
     packages=['ansicon'],
-    package_data={'ansicon': ['*.dll']},
     test_suite='tests',
 
     classifiers=[
-- 
2.42.0.windows.2

